### PR TITLE
Removing regular expression escaping from ValueFilterAdapter

### PR DESCRIPTION
**Pull request by sduskis**
This fixes Issue #447, which shows incorrect behavior for SingleValueColumnFilter.  ValueFilterAdapter is used by SingleValueColumnFilterAdapter, so this fixes behavior for both ValueFilter and SingleValueColumnFilter.
Added a unit test for the scenarios in #447.

### CODE REVIEW

File: bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+import com.google.common.collect.Iterators;
+
+public class TestSingleColumnValueFilter extends AbstractTest {
+
+  byte[] qual = dataHelper.randomData("testSingleValueFilterCompOps");
+  static final int count = 10;
+  static final byte[] COLUMN_FAMILY = Bytes.toBytes("cd");
+  static final byte[] QUALIFIER = Bytes.toBytes("value");
+
+  TableName tableName = IntegrationTests.newTestTableName();
```
Comment (kevinsi4508) [inline bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java]: Is this a shared table? Can we use a new table?

File: bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.Iterators;
+
+public class TestSingleColumnValueFilter extends AbstractTest {
+
+  byte[] qual = dataHelper.randomData("testSingleValueFilterCompOps");
```
Comment (kevinsi4508) [inline bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java]: private static final if possible.

File: bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    }
+    table.close();
+    table = null;
+    result.clear();
+  }
+
+  @Test
+  public void testSingleValueFilterAscii() throws IOException {
+    // Add {, }, and @ to make sure that they do not need to be encoded
+    byte[] rowKeyA = dataHelper.randomData("KeyA{");
```
Comment (kevinsi4508) [inline bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java]: Shouldn't adding data be part of test setup @Before?

File: bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    byte[] rowKeyB = dataHelper.randomData("KeyB}");
+    byte[] rowKeyC = dataHelper.randomData("KeyC@");
+    byte[] qualValueA = dataHelper.randomData("ValueA{");
+    byte[] qualValueB = dataHelper.randomData("ValueB}");
+    byte[] qualValueC = dataHelper.randomData("ValueC@");
+    try (Table table = getConnection().getTable(TABLE_NAME)) {
+      table.put(new Put(rowKeyA).addColumn(COLUMN_FAMILY, qual, qualValueA));
+      table.put(new Put(rowKeyB).addColumn(COLUMN_FAMILY, qual, qualValueC));
+      table.put(new Put(rowKeyC).addColumn(COLUMN_FAMILY, qual, qualValueC));
+      assertSizeOfSCVF(table, 1, CompareOp.EQUAL, qualValueA);
```
Comment (kevinsi4508) [inline bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java]: Add comments and explain why we expect 1 (same for the rest of assertions).

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/ValueFilterAdapter.java [hunk 5]
```diff
@@ -65,27 +63,23 @@ public FilterSupportStatus isFilterSupported(
 
   private RowFilter adaptBinaryComparator(
       CompareOp compareOp, BinaryComparator comparator) throws IOException {
-    byte[] comparatorValue = comparator.getValue();
-    ByteArrayOutputStream baos =
-        new ByteArrayOutputStream(comparatorValue.length * 2);
-    ReaderExpressionHelper.writeQuotedRegularExpression(baos, comparatorValue);
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/ValueFilterAdapter.java]: writeQuotedRegularExpression() is used to escape special characters. I wonder why this is not needed now. Notice that writeQuotedRegularExpression() is used in other places too. If this is a problem, we need to remove it all together.

**Comment by kevinsi4508** [comment]
We do want to escape the characters for equality comparison.

File: bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java (hunk 3 above)
Comment (sduskis) [inline bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java]: I'll move this to TestFilters.  This is a stand alone test condition that doesn't interact with the other @Test methods.

Comment (kevinsi4508) [inline bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java]: I think it is better to have setup data in @Before, then have test cases:
1. testEqual
2. testGreater
3. testLess
...

Comment (kevinsi4508) [inline bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSingleColumnValueFilter.java]: Up to you, it is a nit.

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/ValueFilterAdapter.java (hunk 5 above)
Comment (sduskis) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/ValueFilterAdapter.java]: moved regex escaping to the EQUALS case only

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**